plugins/newsdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,185 @@
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local DataStorage = require("datastorage")
+local FFIUtil = require("ffi/util")
+local util = require("frontend/util")
+local T = FFIUtil.template
+local _ = require("gettext")
+local logger = require("logger")
+local ffi = require("ffi")
+local http = require("socket.http")
+local ltn12 = require("ltn12")
+
+
+local NewsDownloader = WidgetContainer:new{}
+
+local initialized = false  -- for only once lazy initialization
+local FEED_CONFIG_FILE = "feeds.xml"
+local FILE_EXTENSION = ".html"
+local NEWS_DL_DIR_NAME = "news"
+local NEWS_DL_DIR, FEED_CONFIG_PATH
+
+local function deserializeXMLString(xml_str)
+    -- uses LuaXML https://github.com/manoelcampos/LuaXML
+    -- The MIT License (MIT)
+    -- Copyright (c) 2016 Manoel Campos da Silva Filho
+    local treehdl = require("lib/handler")
+    local libxml = require("lib/xml")
+
+    --Instantiate the object the states the XML file as a Lua table
+    local xmlhandler = treehdl.simpleTreeHandler()
+    --Instantiate the object that parses the XML to a Lua table
+    libxml.xmlParser(xmlhandler):parse(xml_str)
+    return xmlhandler.root
+end
+
+local function deserializeXML(filename)
+    logger.dbg("NewsDownloader: File to deserialize: ", filename)
+    local f, e = io.open(filename, "r")
+    if f then
+        local xmltext = f:read("*a")
+        f:close()
+        return deserializeXMLString(xmltext)
+    else
+        logger.warn("NewsDownloader: XML file not found", filename, e)
+    end
+end
+
+function NewsDownloader:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function NewsDownloader:addToMainMenu(menu_items)
+    if not initialized then
+        NEWS_DL_DIR = string.format("%s/%s/", DataStorage:getDataDir(), NEWS_DL_DIR_NAME)
+        if not lfs.attributes(NEWS_DL_DIR, "mode") then
+            lfs.mkdir(NEWS_DL_DIR)
+        end
+
+        FEED_CONFIG_PATH = NEWS_DL_DIR .. FEED_CONFIG_FILE
+        if not lfs.attributes(FEED_CONFIG_PATH, "mode") then
+            logger.dbg("NewsDownloader: Creating init configuration")
+            FFIUtil.copyFile(FFIUtil.joinPath(self.path, FEED_CONFIG_FILE),
+                             FEED_CONFIG_PATH)
+        end
+        initialized = true
+    end
+
+    menu_items.rss_news_downloader = {
+        text = _("Simple News(RSS/Atom) Downloader"),
```

Comment:
Newsfeed isn't odd (at most you might argue it's not yet ready to be spelled as one word),[1] but primarily it's the "Simple" I don't like and the capital in "Downloader" that goes against the UX guidelines.

If you leave it as is and change it later then all of the volunteer translators are going to have to do the work twice. That's not cool, especially for something that'd take only a minute or two to change. :-)

[1]
* https://support.mozilla.org/t5/News-Feeds-Blogs-and-Social/How-to-Subscribe-to-News-Feeds-and-Blogs/ta-p/16473
* https://www.ft.com/news-feed
* http://www.independent.ie/business/technology/facebook-introducing-tools-to-help-users-spot-fake-news-on-their-news-feed-35602736.html
* http://www.newsweek.com/newsfeed
* https://www.ft.com/news-feed